### PR TITLE
Adds validation on destroy user

### PR DESCRIPTION
**Pull request by rodrigoulisses**
With this validation the user will not be allowed to remove his profile.

### CODE REVIEW

File: app/business/destroy_user.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class DestroyUser
+  def initialize(options = {})
```
Comment (dmitryrck) [inline app/business/destroy_user.rb]: ou named parameters #justsaying :P

Comment (rafaelrpbelo) [inline app/business/destroy_user.rb]: prefiro tbm, até melhor de debugar

File: app/business/destroy_user.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class DestroyUser
```
Comment (marclerodrigues) [inline app/business/destroy_user.rb]:
```ruby
class DestroyUser
 def initialize(user, current_user)
  @user = user
  @current_user = current_user
  @errors = {}
 end

 def destroy
  if allowed?
   @user.destroy
  else
   @errors[:alert] = I18n.t('users.notice.destroy.not_permitted')
   false
  end
 end

 private
 
 def allowed?
  @user.id != @current_user.id
 end
end
```

Just a suggestion... IMHO :smile:

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -38,10 +38,8 @@ def update
   end
 
   def destroy
-    user = User.find(params[:id])
```
Comment (marclerodrigues) [inline app/controllers/users_controller.rb]:
```ruby
if @destroied = DestroyUser.destroy!(user: User.find(params[:id]), current_user: current_user)
  flash.now[:notice] = I18n.t('users.notice.destroy.success')
else
 flash.now = @destroied.errors
end

@users = User.all
```

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -38,10 +38,8 @@ def update
   end
 
   def destroy
```
Comment (marclerodrigues) [inline app/controllers/users_controller.rb]: You should have a controller test for this.

**Pull request merged**